pilot/cmd/pilot-agent/main.go

```diff
@@ -177,8 +178,15 @@ var (
 					}
 				}
 			}
-			role.DNSDomain = getDNSDomain(role.DNSDomain)
-			pilotSAN = getPilotSAN(role.DNSDomain, ns)
+
+			// Parse the DNSDomain based upon service registry type into a registry specific domain.
+			DNSDomain = getDNSDomain(DNSDomain)
+
+			// role.ServiceNode() returns a string based upon this META which isn't set in the proxy-init.
+			role.DNSDomains[0] = DNSDomain
```

Comment:
Isn't this a panic?  `role.DNSDomains` is nil.

Reply:
That is probably why this pr is exploding during testing.  I haven't had a chance today to dig into this - but going to start now.  Thanks for spotting this!